feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt

```diff
@@ -152,6 +157,8 @@ internal class AmazonBilling constructor(
         replaceSkuInfo: ReplaceSkuInfo?,
         presentedOfferingIdentifier: String?
     ) {
+        if (checkObserverMode()) return
```

Comment:
i added this check before any interaction with the purchasingservice...based on [docs](https://developer.amazon.com/docs/in-app-purchasing/iap-implement-iap.html#purchasingservice), none of those calls will work without the listener registered